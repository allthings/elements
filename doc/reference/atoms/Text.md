# Text
Text will be used for everywhere a text appears.
The only exception is in molecules that already provide the
text component for you.

```example
<Text>
  Text will be used for everywhere a text appears. The only exception
  is in molecules that already provide the text component for you.
</Text>
```

```example
<Text autoBreak={true}>
  Textwillbeusedforeverywhereatextappears.Theonlyexceptionisinmoleculesthatalreadyprovidethetextcomponentforyou.Textwillbeusedforeverywhereatextappears.Theonlyexceptionisinmoleculesthatalreadyprovidethetextcomponentforyou.
 </Text>
## Usage
| Name        | Type           | Description  |
| ----------- |:--------------:| ------------:|
|align|enum|
|autoBreak|bool|
|block|bool|<br>Default: true
|children|node|
|color|string|<br>Default: 'text'
|italic|bool|
|size|enum|<br>Default: 'l'
|strong|bool|
|underline|bool|
|lineThrough|bool|
