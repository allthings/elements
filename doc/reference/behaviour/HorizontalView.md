<!-- 
This is an auto-generated markdown. 
You can change it in "src/behaviour/HorizontalView.jsx" and run build:docs to update this file.
-->
# HorizontalView
HorizontalView is used to Views in a horizontal row, and will do smooth transitions between them.
The HorizontalView will always show the latest item that get's passed in as a children.

So for this, it will only show the SecondItem.
```
<HorizontalView>
 <FirstItem />
 <SecondItem />
</HorizontalView>
```

To do transitions between the items, you just *pass* in or *remove children*. So if we update the previous example
to be like this, it will transition from the SecondItem to the FirstItem:

```
<HorizontalView>
 <FirstItem />
</HorizontalView>
```

## Using with React Router
The HorizontalView can work perfectly together with React Router. It will allow you
to have smooth transition from one Route to another.

### Example
We are building a taco app, because everyone likes tacos. It consists of 3 screens:
An overview of all available tacos (/tacos), a taco detail page (/tacos/:id) and a list of dips for that are
a good fit with that taco (/tacos/:id/dips).

If you go to `/tacos` only the first route will be matched, the `TacoList` will render.
Now you click a link in the `TacosList` it will bring you to `/tacos/8343`. React Router will
render `TacosList` and `TacosDetail` and HorizontalView do a smooth transition from `TacosList` to
`TacosDetail`.

```
<HorizontalView>
 <Route path="/tacos" component={TacosList}/>
 <Route path="/tacos/:id" component={TacosDetail}/>
 <Route path="/tacos/:id/dips" component={TacosDipsView}/>
</HorizontalView>
```
## Usage
| Name        | Type           | Description  |
| ----------- |:--------------:| ------------:|
|children **(required)**|array|
