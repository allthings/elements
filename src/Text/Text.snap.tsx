// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
.css-0,
[data-css-0] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #333333;
}

.css-1,
[data-css-1] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

<div
  data-css-0=""
  data-css-1=""
/>
`
