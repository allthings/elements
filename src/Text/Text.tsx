import { css } from 'glamor'
import neue from 'neue'
import React from 'react'
import { useTheme } from '../Theme'
import View, { IViewProps } from '../View'

if (typeof window !== `undefined`) {
  neue.load([
    {
      families: ['Open Sans:n3,n3i,n4,n4i,n6,n6i,n7,n7i'],
      css:
        '//fonts.googleapis.com/css?family=Open+Sans:300,300i,400,400i,600,600i,700,700i',
    },
  ])
}

const availableSizes = {
  xs: 10,
  s: 12,
  m: 13,
  l: 14,
  xl: 18,
  xxl: 18,
  giant: 24,
}

export type TextSizeType = number | keyof typeof availableSizes

const textWeightMap = {
  light: 300,
  regular: 400,
  'semi-bold': 600,
  bold: 700,
}

export type TextWeightType = keyof typeof textWeightMap

interface ITextStyles {
  size?: TextSizeType
  block?: boolean
  italic?: boolean
  /**
   * @deprecated
   */
  strong?: boolean
  underline?: boolean
  lineThrough?: boolean
  weight?: TextWeightType
  align?: 'left' | 'center' | 'right'
  autoBreak?: boolean
}

export const createTextStyles = ({
  block = false,
  italic = false,
  strong = false,
  size = 'l',
  weight,
  underline = false,
  lineThrough = false,
  align,
  autoBreak,
}: ITextStyles = {}) =>
  css({
    display: block ? 'block' : 'inline',
    fontFamily: '"Open Sans", Helvetica, Arial, sans-serif',
    fontStyle: italic && 'italic',
    fontWeight: weight ? textWeightMap[weight] : strong ? 600 : 400,
    fontSize: typeof size === 'number' ? size : availableSizes[size],
    textDecoration:
      (underline && 'underline') || (lineThrough && 'line-through'),
    textAlign: align,
    whiteSpace: autoBreak && 'pre-wrap',
    wordBreak: autoBreak && 'break-word',
    WebkitFontSmoothing: 'antialiased',
    MozOsxFontSmoothing: 'grayscale',
  })

export interface IText {
  readonly color?: string
}

export type ITextProps = IText & ITextStyles & IViewProps

let hasWarnedBefore = false

/**
 * Text will be used for everywhere a text appears.
 * The only exception is in molecules that already provide the
 * text component for you.
 *
 * ```example
 * <Text>
 *   Text will be used for everywhere a text appears. The only exception
 *   is in molecules that already provide the text component for you.
 * </Text>
 * ```
 *
 * ```example
 * <Text autoBreak={true}>
 *   Textwillbeusedforeverywhereatextappears.Theonlyexceptionisinmoleculesthatalreadyprovidethetextcomponentforyou.Textwillbeusedforeverywhereatextappears.Theonlyexceptionisinmoleculesthatalreadyprovidethetextcomponentforyou.
 *  </Text>
 *  ```
 */
const Text = ({
  color = 'text',
  block = true,
  children,
  italic,
  size = 'l',
  strong,
  align,
  underline,
  autoBreak,
  lineThrough,
  weight,
  ...props
}: ITextProps) => {
  const { colorize } = useTheme()

  if (strong === true && process.env.NODE_ENV !== 'production') {
    !hasWarnedBefore &&
      console.warn(
        "The property `strong` is deprecated. Please use `weight` instead. (`strong` now corresponds to weight: 'semi-bold')",
      )
    hasWarnedBefore = true
  }

  return (
    <View
      {...css(
        createTextStyles({
          block,
          italic,
          strong,
          size,
          underline,
          lineThrough,
          align,
          autoBreak,
          weight,
        }),
        { color: colorize(color) },
      )}
      {...props}
    >
      {children}
    </View>
  )
}

export default Text
