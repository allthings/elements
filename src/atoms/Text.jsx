import PropTypes from 'prop-types'
import React from 'react'
import { css } from 'glamor'
import View from '../atoms/View'
import Theme from '../behaviour/Theme'
import WebFont from 'webfontloader'

if (typeof window !== `undefined`) {
  WebFont.load({
    classes: false,
    google: {
      families: ['Open Sans:300,600'],
    },
  })
}

const availableSizes = {
  xs: 10,
  s: 12,
  m: 13,
  l: 14,
  xl: 18,
  xxl: 18,
  giant: 24,
}

/**
 * Text will be used for everywhere a text appears.
 * The only exception is in molecules that already provide the
 * text component for you.
 */
class Text extends React.Component {
  static propTypes = {
    align: PropTypes.oneOf(['left', 'center', 'right']),
    autoBreak: PropTypes.bool,
    block: PropTypes.bool,
    children: PropTypes.node,
    color: PropTypes.string,
    italic: PropTypes.bool,
    size: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl', 'giant']),
    strong: PropTypes.bool,
    underline: PropTypes.bool,
    lineThrough: PropTypes.bool,
  }

  static defaultProps = {
    size: 'l',
    block: true,
    color: 'text',
  }

  render() {
    const {
      color,
      block,
      children,
      italic,
      size,
      strong,
      align,
      underline,
      autoBreak,
      lineThrough,
      ...props
    } = this.props

    const styles = css({
      display: block ? 'block' : 'inline',
      fontFamily: '"Open Sans", Helvetica, Arial, sans-serif',
      fontStyle: italic && 'italic',
      fontWeight: strong && '600',
      fontSize: availableSizes[size],
      textDecoration:
        (underline && 'underline') || (lineThrough && 'line-through'),
      textAlign: align,
      whiteSpace: autoBreak && 'pre-wrap',
      wordBreak: autoBreak && 'break-word',
      WebkitFontSmoothing: 'antialiased',
      MozOsxFontSmoothing: 'grayscale',
    })

    return (
      <Theme>
        {({ colorize }) => (
          <View {...css(styles, { color: colorize(color) })} {...props}>
            {children}
          </View>
        )}
      </Theme>
    )
  }
}

export default Text
