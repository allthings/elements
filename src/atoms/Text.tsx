import React from 'react'
import { css } from 'glamor'
import View, { IView } from '../atoms/View'
import Theme, { IThemeChildren } from '../behaviour/Theme'
import neue from 'neue'

if (typeof window !== `undefined`) {
  neue.load([
    {
      families: ['Open Sans:n4,n4i,n6,n6i'],
      css: '//fonts.googleapis.com/css?family=Open+Sans:400,400i,600,600i',
    },
  ])
}

type sizeType = 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl' | 'giant'

const availableSizes = {
  xs: 10,
  s: 12,
  m: 13,
  l: 14,
  xl: 18,
  xxl: 18,
  giant: 24,
}

interface ITextStyles {
  size: sizeType
  block?: boolean
  italic?: boolean
  strong?: boolean
  underline?: boolean
  lineThrough?: boolean
  align?: 'left' | 'center' | 'right'
  autoBreak?: boolean
}

export const createTextStyles = ({
  block = false,
  italic = false,
  strong = false,
  size = 'm',
  underline = false,
  lineThrough = false,
  align,
  autoBreak,
}: ITextStyles) => {
  return css({
    display: block ? 'block' : 'inline',
    fontFamily: '"Open Sans", Helvetica, Arial, sans-serif',
    fontStyle: italic && 'italic',
    fontWeight: strong && '600',
    fontSize: availableSizes[size],
    textDecoration:
      (underline && 'underline') || (lineThrough && 'line-through'),
    textAlign: align,
    whiteSpace: autoBreak && 'pre-wrap',
    wordBreak: autoBreak && 'break-word',
    WebkitFontSmoothing: 'antialiased',
    MozOsxFontSmoothing: 'grayscale',
  })
}

export interface IText {
  color: string
}

/**
 * Text will be used for everywhere a text appears.
 * The only exception is in molecules that already provide the
 * text component for you.
 *
 * ```example
 * <Text>
 *   Text will be used for everywhere a text appears. The only exception
 *   is in molecules that already provide the text component for you.
 * </Text>
 * ```
 *
 * ```example
 * <Text autoBreak={true}>
 *   Textwillbeusedforeverywhereatextappears.Theonlyexceptionisinmoleculesthatalreadyprovidethetextcomponentforyou.Textwillbeusedforeverywhereatextappears.Theonlyexceptionisinmoleculesthatalreadyprovidethetextcomponentforyou.
 *  </Text>
 *  ```
 */
const Text = ({
  color = 'text',
  block = true,
  children,
  italic,
  size = 'l',
  strong,
  align,
  underline,
  autoBreak,
  lineThrough,
  ...props
}: IText & ITextStyles & IView) => (
  <Theme>
    {({ colorize }: IThemeChildren) => (
      <View
        {...css(
          createTextStyles({
            block,
            italic,
            strong,
            size,
            underline,
            lineThrough,
            align,
            autoBreak,
          }),
          { color: colorize(color) }
        )}
        {...props}
      >
        {children}
      </View>
    )}
  </Theme>
)

export default Text
