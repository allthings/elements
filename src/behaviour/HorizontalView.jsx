import React from 'react'
import PropTypes from 'prop-types'
import View from '../atoms/View'
import { css } from 'glamor'

/**
 * HorizontalView is used to Views in a horizontal row, and will do smooth transitions between them.
 * The HorizontalView will always show the latest item that get's passed in as a children.
 *
 * So for this, it will only show the SecondItem.
 * ```
 * <HorizontalView>
 *  <FirstItem />
 *  <SecondItem />
 * </HorizontalView>
 * ```
 *
 * To do transitions between the items, you just *pass* in or *remove children*. So if we update the previous example
 * to be like this, it will transition from the SecondItem to the FirstItem:
 *
 * ```
 * <HorizontalView>
 *  <FirstItem />
 * </HorizontalView>
 * ```
 *
 * ## Using with React Router
 * The HorizontalView can work perfectly together with React Router. It will allow you
 * to have smooth transition from one Route to another.
 *
 * ### Example
 * We are building a taco app, because everyone likes tacos. It consists of 3 screens:
 * An overview of all available tacos (/tacos), a taco detail page (/tacos/:id) and a list of dips for that are
 * a good fit with that taco (/tacos/:id/dips).
 *
 * If you go to `/tacos` only the first route will be matched, the `TacoList` will render.
 * Now you click a link in the `TacosList` it will bring you to `/tacos/8343`. React Router will
 * render `TacosList` and `TacosDetail` and HorizontalView do a smooth transition from `TacosList` to
 * `TacosDetail`.
 *
 * ```
 * <HorizontalView>
 *  <Route path="/tacos" component={TacosList}/>
 *  <Route path="/tacos/:id" component={TacosDetail}/>
 *  <Route path="/tacos/:id/dips" component={TacosDipsView}/>
 * </HorizontalView>
 * ```
 **/
class HorizontalView extends React.Component {
  static propTypes = {
    children: PropTypes.array.isRequired,
  }

  constructor(props, context) {
    super(props, context)

    const children = props.children
    const currentChild = children.length

    this.state = {
      children,
      currentChild,
      nextChildren: null,
      waitForTransitionEnd: false,
    }
  }

  static getDerivedStateFromProps(
    { children: nextProps },
    { children: previousState }
  ) {
    const [oldChildren, nextChildren] = [previousState, nextProps].map(
      children => children.filter(child => React.isValidElement(child))
    )

    return nextChildren.length < oldChildren.length
      ? {
          nextChildren,
          waitForTransitionEnd: true,
          currentChild: nextChildren.length,
        }
      : {
          children: nextChildren,
          currentChild: nextChildren.length,
        }
  }

  handleTransitionEnd = () => {
    if (this.state.waitForTransitionEnd) {
      this.setState({
        children: this.state.nextChildren,
        nextChildren: null,
        waitForTransitionEnd: false,
      })
    }
  }

  render() {
    const { currentChild, children } = this.state
    const { children: propsChildren, ...props } = this.props
    const translateX = currentChild * -100

    return (
      <View flex="flex" {...css({ overflow: 'hidden' })}>
        <View
          direction="row"
          flex="flex"
          {...css({
            transform: `translate3d(${translateX}%, 0, 0)`,
            transition: '.5s',
          })}
          onTransitionEnd={this.handleTransitionEnd}
          {...props}
        >
          {React.Children.map(children, (child, i) => (
            <View
              // eslint-disable-next-line
              key={i}
              {...css({ width: '100%' })}
              flex="none"
              direction="column"
            >
              {child}
            </View>
          ))}
        </View>
      </View>
    )
  }
}

export default HorizontalView
