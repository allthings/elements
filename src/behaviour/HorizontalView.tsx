import React, { Component, ReactNode } from 'react'
import View, { IViewProps } from '../atoms/View'
import { css } from 'glamor'

interface IHorizontalViewProps {
  children: ReactNode[]
}
interface IState {
  children?: ReactNode[]
  currentChild: number
  nextChildren?: ReactNode[]
  waitForTransitionEnd: boolean
}

/**
 * HorizontalView is used to Views in a horizontal row, and will do smooth transitions between them.
 * The HorizontalView will always show the latest item that get's passed in as a children.
 *
 * So for this, it will only show the SecondItem.
 * ```
 * <HorizontalView>
 *  <FirstItem />
 *  <SecondItem />
 * </HorizontalView>
 * ```
 *
 * To do transitions between the items, you just *pass* in or *remove children*. So if we update the previous example
 * to be like this, it will transition from the SecondItem to the FirstItem:
 *
 * ```
 * <HorizontalView>
 *  <FirstItem />
 * </HorizontalView>
 * ```
 *
 * ## Using with React Router
 * The HorizontalView can work perfectly together with React Router. It will allow you
 * to have smooth transition from one Route to another.
 *
 * ### Example
 * We are building a taco app, because everyone likes tacos. It consists of 3 screens:
 * An overview of all available tacos (/tacos), a taco detail page (/tacos/:id) and a list of dips for that are
 * a good fit with that taco (/tacos/:id/dips).
 *
 * If you go to `/tacos` only the first route will be matched, the `TacoList` will render.
 * Now you click a link in the `TacosList` it will bring you to `/tacos/8343`. React Router will
 * render `TacosList` and `TacosDetail` and HorizontalView do a smooth transition from `TacosList` to
 * `TacosDetail`.
 *
 * ```
 * <HorizontalView>
 *  <Route path="/tacos" component={TacosList}/>
 *  <Route path="/tacos/:id" component={TacosDetail}/>
 *  <Route path="/tacos/:id/dips" component={TacosDipsView}/>
 * </HorizontalView>
 * ```
 **/
class HorizontalView extends Component<
  IViewProps & IHorizontalViewProps,
  IState
> {
  constructor(props: IHorizontalViewProps) {
    super(props)

    const children = props.children
    const currentChild = React.Children.count(children)

    this.state = {
      children,
      currentChild,
      nextChildren: undefined,
      waitForTransitionEnd: false,
    }
  }

  static getDerivedStateFromProps(
    { children: nextProps }: IHorizontalViewProps,
    { children: previousState }: IState
  ) {
    const [oldChildren, nextChildren] = [previousState, nextProps].map(
      (children = []) => children.filter(child => React.isValidElement(child))
    )

    return nextChildren.length < oldChildren.length
      ? {
          nextChildren,
          waitForTransitionEnd: true,
          currentChild: nextChildren.length,
        }
      : {
          children: nextChildren,
          currentChild: nextChildren.length,
        }
  }

  handleTransitionEnd = () => {
    if (this.state.waitForTransitionEnd) {
      this.setState({
        children: this.state.nextChildren,
        nextChildren: undefined,
        waitForTransitionEnd: false,
      })
    }
  }

  render() {
    const { currentChild, children } = this.state
    const { children: propsChildren, ...props } = this.props
    const translateX = (currentChild - 1) * -100

    return (
      <View flex="flex" {...css({ overflow: 'hidden' })}>
        <View
          direction="row"
          flex="flex"
          {...css({
            transform: `translate3d(${translateX}%, 0, 0)`,
            transition: '.5s',
          })}
          onTransitionEnd={this.handleTransitionEnd}
          {...props}
        >
          {React.Children.map(children, (child: ReactNode, i: number) => (
            <View
              // eslint-disable-next-line
              key={i}
              {...css({ width: '100%' })}
              flex="none"
              direction="column"
            >
              {child}
            </View>
          ))}
        </View>
      </View>
    )
  }
}

export default HorizontalView
