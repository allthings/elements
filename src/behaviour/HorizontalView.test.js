import React from 'react'
import HorizontalView from './HorizontalView'
import Text from '../atoms/Text'

describe('HorizontalView behaviors', () => {
  it('should render non-arrays', () => {
    const condition = false
    const wrapper = shallow(
      <HorizontalView id="horizontal">
        <Text id="first" key="first">
          Text 1
        </Text>
        <Text id="second" key="second">
          Text 2
        </Text>
        {undefined}
      </HorizontalView>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('#horizontal').children()).toHaveLength(2)
    expect(wrapper.find('#first').exists()).toEqual(true)
    expect(wrapper.find('#third').exists()).toEqual(false)
    wrapper.unmount()
  })
  it('should render with array too', () => {
    const wrapper = shallow(
      <HorizontalView>
        {[
          <Text id="first" key="first">
            Text 1
          </Text>,
          <Text id="second" key="second">
            Text 2
          </Text>,
          undefined,
        ]}
      </HorizontalView>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('#second').exists()).toEqual(true)
    expect(wrapper.find('#first').exists()).toEqual(false)
    wrapper.unmount()
  })
})
