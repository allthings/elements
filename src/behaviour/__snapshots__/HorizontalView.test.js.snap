// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalView behaviors should render non-arrays 1`] = `
.css-0,
[data-css-0] {
  overflow: hidden;
}

.css-1,
[data-css-1] {
  transform: translate3d(-200%, 0, 0);
  transition: .5s;
  -webkit-transform: translate3d(-200%, 0, 0);
  -webkit-transition: .5s;
  -moz-transition: .5s;
}

.css-2,
[data-css-2] {
  width: 100%;
}

<View
  alignH="start"
  alignV="stretch"
  data-css-0=""
  fill={false}
  flex="flex"
  onRef={[Function]}
>
  <View
    alignH="start"
    alignV="stretch"
    data-css-1=""
    direction="row"
    fill={false}
    flex="flex"
    id="horizontal"
    onRef={[Function]}
    onTransitionEnd={[Function]}
  >
    <View
      alignH="start"
      alignV="stretch"
      data-css-2=""
      direction="column"
      fill={false}
      flex="none"
      key="0/.$first"
      onRef={[Function]}
    >
      <Text
        block={true}
        color="text"
        id="first"
        key="first"
        size="l"
      >
        Text 1
      </Text>
    </View>
    <View
      alignH="start"
      alignV="stretch"
      data-css-2=""
      direction="column"
      fill={false}
      flex="none"
      key="1/.$second"
      onRef={[Function]}
    >
      <Text
        block={true}
        color="text"
        id="second"
        key="second"
        size="l"
      >
        Text 2
      </Text>
    </View>
    <View
      alignH="start"
      alignV="stretch"
      data-css-2=""
      direction="column"
      fill={false}
      flex="none"
      key="2/.2"
      onRef={[Function]}
    />
  </View>
</View>
`;

exports[`HorizontalView behaviors should render with array too 1`] = `
.css-0,
[data-css-0] {
  overflow: hidden;
}

.css-1,
[data-css-1] {
  transform: translate3d(-200%, 0, 0);
  transition: .5s;
  -webkit-transform: translate3d(-200%, 0, 0);
  -webkit-transition: .5s;
  -moz-transition: .5s;
}

.css-2,
[data-css-2] {
  width: 100%;
}

<View
  alignH="start"
  alignV="stretch"
  data-css-0=""
  fill={false}
  flex="flex"
  onRef={[Function]}
>
  <View
    alignH="start"
    alignV="stretch"
    data-css-1=""
    direction="row"
    fill={false}
    flex="flex"
    onRef={[Function]}
    onTransitionEnd={[Function]}
  >
    <View
      alignH="start"
      alignV="stretch"
      data-css-2=""
      direction="column"
      fill={false}
      flex="none"
      key="0/.$first"
      onRef={[Function]}
    >
      <Text
        block={true}
        color="text"
        id="first"
        key="first"
        size="l"
      >
        Text 1
      </Text>
    </View>
    <View
      alignH="start"
      alignV="stretch"
      data-css-2=""
      direction="column"
      fill={false}
      flex="none"
      key="1/.$second"
      onRef={[Function]}
    >
      <Text
        block={true}
        color="text"
        id="second"
        key="second"
        size="l"
      >
        Text 2
      </Text>
    </View>
    <View
      alignH="start"
      alignV="stretch"
      data-css-2=""
      direction="column"
      fill={false}
      flex="none"
      key="2/.2"
      onRef={[Function]}
    />
  </View>
</View>
`;
